Fix: Improve error message for invalid template variables in virtualenvs.prompt

JaviMaligno opened the pull request:
Fixes #10635
Closes #10649

## Summary by Sourcery

Improve validation and error reporting for virtual environment prompt templates when creating virtualenvs.

Bug Fixes:
- Raise a clear PoetryRuntimeError when the virtualenvs.prompt template contains unknown variables instead of failing with a generic KeyError.

Tests:
- Add a regression test verifying that an invalid virtualenvs.prompt template variable produces a descriptive error message including the invalid name and config key.

radoering commented on tests/utils/env/test_env_manager.py:
```diff
@@ -1302,3 +1306,38 @@ def test_generate_env_name_uses_real_path(
+    manager: EnvManager,
+    poetry: Poetry,
+    config: Config,
+    mocker: MockerFixture,
```
`mocker` seems to be unused.

radoering commented on tests/utils/env/test_env_manager.py:
```diff
@@ -1302,3 +1306,38 @@ def test_generate_env_name_uses_real_path(
+) -> None:
+    config.merge({"virtualenvs": {"prompt": "{project_name}-{invalid_var}"}})
+
+    mocked_python_register("3.9.0", make_system=True)
```
We do not need `mocked_python_register,` do we?

JaviMaligno commented:
Thanks for the review feedback! I've addressed all the comments:

**Changes in this commit:**

1. **Exception type changed**: Replaced `PoetryRuntimeError` with `PoetryConsoleError` so error messages now display in red (as `PoetryConsoleError` is handled by cleo's standard exception handler).

2. **Removed unused test parameters**: Removed the unused `mocker: MockerFixture` parameter from both test functions.

3. **Pre-commit fixes**: Ran ruff which fixed 5 linting errors (removed unused imports: `Config`, `NullIO`, `virtualenv`) and reformatted the code.

Let me know if there's anything else that needs to be addressed!
